newhelm/typed_data.py

```diff
@@ -0,0 +1,34 @@
+from typing import Any, Dict, Type, TypeVar
+from typing_extensions import Self
+from pydantic import BaseModel
+
+
+_BaseModelType = TypeVar("_BaseModelType", bound=BaseModel)
+
+
+class TypedData(BaseModel):
+    """This is a generic container that allows Pydantic to do polymorphic serialization.
+
+    This is useful in situations where you have an unknown set of classes that could be
+    used in a particular field.
+    """
+
+    type: str
```

Comment:
Note that serializing the fully-qualified name means that users cannot move `TypedData` classes between modules.